mlflow/server/js/src/experiment-tracking/pages/experiment-evaluation-datasets/components/ExportTracesToDatasetModal.tsx

```diff
@@ -0,0 +1,216 @@
+import {
+  Checkbox,
+  Empty,
+  Input,
+  Modal,
+  SearchIcon,
+  Table,
+  TableCell,
+  TableHeader,
+  TableRow,
+  TableSkeletonRows,
+  useDesignSystemTheme,
+} from '@databricks/design-system';
+import { ColumnDef, flexRender, getCoreRowModel, useReactTable } from '@tanstack/react-table';
+import { FormattedMessage } from 'react-intl';
+import { useInfiniteScrollFetch } from '../hooks/useInfiniteScrollFetch';
+import { useSearchEvaluationDatasets } from '../hooks/useSearchEvaluationDatasets';
+import { EvaluationDataset } from '../types';
+import { getTrace } from '../../../utils/TraceUtils';
+import { useCallback, useEffect, useState } from 'react';
+import { getModelTraceId, ModelTrace } from '@mlflow/mlflow/src/shared/web-shared/model-trace-explorer';
+import { compact } from 'lodash';
+import { extractDatasetInfoFromTraces } from '../utils/datasetUtils';
+import { useUpsertDatasetRecordsMutation } from '../hooks/useUpsertDatasetRecordsMutation';
+import { CreateEvaluationDatasetButton } from './CreateEvaluationDatasetButton';
+
+const CheckboxCell: ColumnDef<EvaluationDataset, string>['cell'] = ({ row }) => {
+  return (
+    <Checkbox
+      componentId="mlflow.export-traces-to-dataset-modal.checkbox"
+      isChecked={row.getIsSelected()}
+      onChange={row.getToggleSelectedHandler()}
+    />
+  );
+};
+
+const columns: ColumnDef<EvaluationDataset, string>[] = [
+  {
+    id: 'checkbox',
+    header: '',
+    size: 32,
+    maxSize: 32,
+    cell: CheckboxCell,
+  },
+  {
+    id: 'name',
+    accessorKey: 'name',
+    header: 'Name',
+  },
+];
+
+export const ExportTracesToDatasetModal = ({
+  experimentId,
+  visible,
+  setVisible,
+  selectedTraceInfos,
+}: {
+  experimentId: string;
+  visible: boolean;
+  setVisible: (visible: boolean) => void;
+  selectedTraceInfos: ModelTrace['info'][];
+}) => {
+  const { theme } = useDesignSystemTheme();
+  const [isLoadingTraces, setIsLoadingTraces] = useState(true);
+  const [datasetRowsToExport, setDatasetRowsToExport] = useState<any[]>([]);
+  const [searchFilter, setSearchFilter] = useState('');
+  const [internalSearchFilter, setInternalSearchFilter] = useState(searchFilter);
+
+  // we need to fetch trace data, as the dataset rows
+  // require the inputs of the root span
+  useEffect(() => {
```

Comment:
maybe extract it to a separate hook to be reused/self-contained?
I'd also suggest to wrap it (even entire Promise.all) in tanstack's `useQuery` so loading state/serrors etc. are way easier to handle